../src/log.c:213:9: error: use of undeclared identifier 'Dl_info'
        Dl_info info = {
        ^
../src/log.c:219:9: error: implicit declaration of function 'dladdr' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
        dladdr((void *) (uintptr_t) ip, &info);
        ^
../src/log.c:219:42: error: use of undeclared identifier 'info'
        dladdr((void *) (uintptr_t) ip, &info);
                                         ^
../src/log.c:221:75: error: use of undeclared identifier 'info'; did you mean 'int'?
               depth++, ip, symbol, off, info.dli_fname, ip - (uintptr_t) info.dli_fbase);
                                                                          ^~~~
                                                                          int
../src/log.c:221:75: error: expected expression
../src/log.c:221:42: error: use of undeclared identifier 'info'
               depth++, ip, symbol, off, info.dli_fname, ip - (uintptr_t) info.dli_fbase);
                                         ^

--- meson.build.orig	2023-03-13 12:07:23 UTC
+++ meson.build
@@ -312,7 +312,7 @@ c_opts = [
 cxx = meson.get_compiler('cpp')
 
 c_opts = [
-  '-D_ISOC99_SOURCE', '-D_ISOC11_SOURCE', '-D_GNU_SOURCE', '-D_XOPEN_SOURCE=700',
+  '-D_ISOC99_SOURCE', '-D_ISOC11_SOURCE', '-D_GNU_SOURCE',
   '-U__STRICT_ANSI__', '-Wmissing-prototypes',
 
   # Warnings to ignore
